internal/core/attribute.cpp

```diff
@@ -239,6 +239,12 @@ void calc_attributes(
     AttributeMap::AttributeComputeParam params;
 
     for (std::size_t i = from; i < to; ++i) {
+        auto above = primary.value(i) - top.value(i);
+        auto below = bottom.value(i) - primary.value(i);
+
+        src_window.move(above, below);
+        dst_window.move(above, below);
```

Comment:
Now that we have a top and bottom surface it seams like the current design for windows (using above below) is wrong, and actually complicates things.  Instead what makes more sense now is looking at the window as a range, from some depth to some depth:

```
const float from = top.value(i);
const float to   = bottom.value(i);

auto window = Window(from, to);
```

This (together with striding and margins, like before) gives us everything we need.

This also removes the need to pass in 3 surfaces

Reply:
> You still need a reference point. Remember that we need to recalculate sampling rate from SOME point. So you always want you 3 surfaces.

You need _a_ reference point, not necessarily the middle point. We used the point from the input surface because that was the only point we had before. Now you can choose either for the 3 point (top, ref, bottom).

Reply:
No!! :scream_cat: 

We used it as a reference point because that was the only reference point that made sense! We discussed it!!

Why would we change the behavior to something wrong/more complicated/inconsistent with how horizon works now? The idea of this PR was to make minimal changes to allow the Attributes-between-surfaces case, but still keeping horizon intact.

Top and bottom surface and align horizontally, but not as per stepsize, not supplied, not the one from the file.

If your file stepsize is 4, your above is 7 and you new samplesize is 3, there is a big difference which point would be a reference one. For example when interpolating you would get two totally different results if you
- set reference point to main surface. Then you interpolate data to depth, depth - 3, depth - 6.
- set reference point to top surface (aka main + above). Then you interpolate data to top, top - 3, top - 6. You never get data from your supplied surface point as it is top - 7.

Obviously we want to take reference points from primary surface supplied by user. So if they supply a point that has data in the file we return as samplevalue that exact data, and not some estimate.

Reply:
You are right, I forgot about unalignement of input above / below. This is why I don't like to unify things to early, I'm to stupid to keep all the possible conditions and outcomes in mind at the same time